lib/fetch/index.js

```diff
@@ -555,7 +555,7 @@ async function mainFetch (fetchParams, recursive = false) {
 
   // 8. If request’s referrer is not "no-referrer", then set request’s
   // referrer to the result of invoking determine request’s referrer.
-  if (request.referrer !== 'no-referrer') {
+  if (request[kState].referrer !== 'no-referrer') {
```

Comment:
Bug?

This is maybe not correct, but the issue is, that we have a custom getter for request.referrer. So if request.referrer is 'no-referrer' than request.referrer returns empty string (''). 

https://github.com/nodejs/undici/blob/3372aed902951f135d710d0fb430f39b8571e547/lib/fetch/request.js#L609

Reply:
this should be reverted